### PR TITLE
Support enabling SSH daemon

### PR DESCRIPTION
**Pull request by saimonation**
@toddatctera: FYI. This won't SSH to the filer and perform the challenge response (this will be done manually), but it does offer a way to initialize the SSH daemon using a public key (either provided by the user, or auto generated by the SDK)

### CODE REVIEW

File: cterasdk/edge/ssh.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        param = Object()
+
+        private_key = None
+        if not public_key:
+            private_key = rsa.generate_private_key(public_exponent=exponent, key_size=key_size)
+            public_key = private_key.public_key().public_bytes(Encoding.OpenSSH, PublicFormat.OpenSSH)
+
+        param.publicKey = public_key.decode('utf-8')
+
+        if private_key:
```
Comment (ygalblum) [inline cterasdk/edge/ssh.py:32]: private_key is set only inside the previous if statement context. Why not have this code there? Plus, I'd move the actual save operation into a separate method (just to make the code cleaner)

**Comment by ygalblum** [comment]
@saimonation In addition, I would add an option to pass a path to a local file which the SDK then reads and passes its content to the Gateway

File: cterasdk/edge/ssh.py (hunk 1 above)
Comment (saimonation) [reply cterasdk/edge/ssh.py:32]: You're right. It bothered me too. Checkout this commit.. I like it better. I'd use the crypto module to:
1) Better certificate management when initializing requests (so we can avoid the warning message we always get)
2) Verify PEM files provided by the user when updating the filer's cert in `edge/ssl.py`

File: cterasdk/lib/crypto.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+        logging.getLogger().info('Saving public key.')
+        path = filesystem.save(dirpath, '{}.pub'.format(filename), self.public_key)
+        logging.getLogger().info('Saved public key. %s', {'filepath': path, 'format': 'OpenSSH'})
+
+
+class CryptoServices:
+
+    @staticmethod
+    def generate_rsa_key_pair(exponent=65537, key_size=2048):
```
Comment (ygalblum) [inline cterasdk/lib/crypto.py:37]: I would go further and offer an API that generates the pair, saves it and returns the public key

**Review by ygalblum — APPROVED** [review]

**Comment by toddatctera** [comment]
> 
> 
> @toddatctera: FYI. This won't SSH to the filer and perform the challenge response (this will be done manually), but it does offer a way to initialize the SSH daemon using a public key (either provided by the user, or auto generated by the SDK)

Sounds great to me. Appreciate the effort from both of you. 
I wouldn't expect our SDK to implement it's own SSH client anyway, but I think being able to generate a keypair if a public key isn't provided will be awesome.

**Pull request merged**